primitive/message.go

```diff
@@ -85,28 +85,33 @@ func NewMessage(topic string, body []byte) *Message {
 	return msg
 }
 
-// SetDelayTimeLevel set message delay time to consume.
+// WithDelayTimeLevel set message delay time to consume.
 // reference delay level definition: 1s 5s 10s 30s 1m 2m 3m 4m 5m 6m 7m 8m 9m 10m 20m 30m 1h 2h
 // delay level starts from 1. for example, if we set param level=1, then the delay time is 1s.
-func (msg *Message) SetDelayTimeLevel(level int) {
+func (msg *Message) WithDelayTimeLevel(level int) *Message {
```

Comment:
modify the API signature should be more careful. IMO, it is better to add a new API here and leave the old one with Deprecated flag and then remove it.

Reply:
becase of the alpha now, so maybe okay here. when we release, we should be careful.